accelerator/core/mam_core.h

```diff
@@ -25,81 +25,109 @@ extern "C" {
- * @param chid - Channel ID[out]
- * @param epid - Endpoint ID[out]
- * @param bundle - The bundle contains the Header and Packets of the current Message[out]
- * @param msg_id - Unique Message identifier under each channel[out]
- * @param ch_remain_sk - The number of MSS channel secret keys that haven't been used[out]
- * @param ep_remain_sk - The number of MSS endpoint secret keys of that haven't been used[out]
+ * @param service[in] IRI node end point service
+ * @param api[in,out] The API
+ * @param channel_depth[in] Depth of channel merkle tree
+ * @param psks[in] Pre-Shared Key set
+ * @param ntru_pks[in] NTRU public key set
+ * @param payload[in] The message that is going to send with MAM
+ * @param chid[out] Channel ID
+ * @param bundle[out] The bundle contains the Header and Packets of the current Message
+ * @param msg_id[out] Unique Message identifier under each channel
  *
  * @return return code
  */
 status_t ta_mam_written_msg_to_bundle(const iota_client_service_t* const service, mam_api_t* const api,
-                                      const size_t channel_depth, const size_t endpoint_depth, tryte_t* const chid,
-                                      tryte_t* const epid, mam_psk_t_set_t psks, mam_ntru_pk_t_set_t ntru_pks,
-                                      char const* const payload, bundle_transactions_t** bundle, tryte_t* msg_id,
-                                      uint32_t* ch_remain_sk, uint32_t* ep_remain_sk);
+                                      const size_t channel_depth, tryte_t* const chid, mam_psk_t_set_t psks,
+                                      mam_ntru_pk_t_set_t ntru_pks, char const* const payload,
+                                      bundle_transactions_t** bundle, tryte_t* msg_id,
+                                      mam_mss_key_status_t* key_status);
 
 /**
  * @brief Write an announcement to bundle.
  *
  * Write the announcement of the next Channel ID (chid1) or Endpoint ID (epid1). This function would
  * automatically determine which one is going to be generated.
  *
- * @param api - The API [in,out]
- * @param channel_depth - Depth of channel merkle tree[in]
- * @param endpoint_depth - Depth of endpoint merkle tree[in]
- * @param chid - Channel ID[in]
- * @param ch_remain_sk - The number of MSS channel secret keys that haven't been used[in]
- * @param ep_remain_sk - The number of MSS endpoint secret keys of that haven't been used[in]
- * @param psks - Pre-Shared Key set[in]
- * @param ntru_pks - NTRU public key set[in]
- * @param chid1 - The next Channel ID[in]
- * @param bundle - The bundle contains the Header and Packets of the current Message[out]
+ * @param api[in,out] The API
+ * @param channel_depth[in] Depth of channel merkle tree
+ * @param chid[in] Channel ID
+ * @param ch_remain_sk[in] The number of MSS channel secret keys that haven't been used
+ * @param psks[in] Pre-Shared Key set
+ * @param ntru_pks[in] NTRU public key set
+ * @param chid1[in] The next Channel ID
```

Comment:
Can we have a more intuitive naming?
Such as `chid_next`?

Reply:
chid1 is the naming in MAM spec, but we can change to `chid_next` if needed

Reply:
Let us follow the MAM spec.